CRM/Contribute/Form/Contribution/ThankYou.php

```diff
@@ -51,8 +51,9 @@ public function preProcess(): void {
     $this->assign('initial_reminder_day', CRM_Utils_Array::value('initial_reminder_day', $this->_values));
     // Link (button) for users to create their own Personal Campaign page
     if ($linkText = CRM_PCP_BAO_PCP::getPcpBlockStatus($this->getContributionPageID(), 'contribute')) {
+      $contributionPageID = $this->getContributionPageID();
       $linkTextUrl = CRM_Utils_System::url('civicrm/contribute/campaign',
-        "action=add&reset=1&pageId=' . $this->getContributionPageID() . '&component=contribute",
+        "action=add&reset=1&pageId=' . $contributionPageID . '&component=contribute",
```

Comment:
Is this change necessary?

Reply:
@mattwire something weird was going on - it was giving me errors & no matter how I stared it didn't go away until I did this

Reply:
OK - sleep helped https://github.com/civicrm/civicrm-core/pull/28374